modules/rtbhouseBidAdapter.js

```diff
@@ -96,30 +63,257 @@ export const spec = {
     };
   },
   interpretResponse: function (serverResponse, originalRequest) {
-    serverResponse = serverResponse.body;
-    const bids = [];
-
-    if (utils.isArray(serverResponse)) {
-      serverResponse.forEach(serverBid => {
-        if (serverBid.price !== 0) {
-          const bid = {
-            requestId: serverBid.impid,
-            mediaType: BANNER,
-            cpm: serverBid.price,
-            creativeId: serverBid.adid,
-            ad: serverBid.adm,
-            width: serverBid.w,
-            height: serverBid.h,
-            ttl: 55,
-            netRevenue: true,
-            currency: 'USD'
-          };
-          bids.push(bid);
-        }
-      });
+    const responseBody = serverResponse.body;
+    if (!utils.isArray(responseBody)) {
+      return [];
     }
+
+    const bids = [];
+    responseBody.forEach(serverBid => {
+      if (serverBid.price === 0) {
+        return;
+      }
+      if (serverBid.adm.startsWith('{"native')) {
+        bids.push(interpretNativeBid(serverBid));
+      } else {
+        bids.push(interpretBannerBid(serverBid));
+      }
+    });
     return bids;
   }
 };
 
+function buildEndpointUrl(region) {
+  return 'https://' + region + '.' + ENDPOINT_URL;
+}
+
+/**
+ * Produces OpenRTB Imp object from a slot config
+ */
+function mapImpression(slot) {
+  return {
+    id: slot.bidId,
+    banner: mapBanner(slot),
+    native: mapNative(slot),
+    tagid: slot.adUnitCode.toString()
+  };
+}
+
+/**
+ * Produces OpenRTB Banner object
+ */
+function mapBanner(slot) {
+  if (slot.mediaType === 'banner' ||
+    utils.deepAccess(slot, 'mediaTypes.banner') ||
+    (!slot.mediaType && !slot.mediaTypes)) {
+    return {
+      w: slot.sizes[0][0],
+      h: slot.sizes[0][1],
+      format: slot.sizes.map(size => ({
+        w: size[0],
+        h: size[1]
+      }))
+    };
+  }
+}
+
+/**
+ * Produces an OpenRTB Site object
+ */
+function mapSite(validRequest) {
+  const pubId = validRequest && validRequest.length > 0
+    ? validRequest[0].params.publisherId : 'unknown';
```

Comment:
End commit oid is not part of the pull request

Reply:
Fixed.